crates/compiler/types/src/subs.rs

```diff
@@ -4707,6 +4987,34 @@ fn storage_copy_var_to_help(env: &mut StorageCopyVarToEnv<'_>, var: Variable) ->
                     Record(record_fields, storage_copy_var_to_help(env, ext))
                 }
 
+                Tuple(elems, ext) => {
+                    let tuple_elems = {
+                        let new_variables =
+                            VariableSubsSlice::reserve_into_subs(env.target, elems.len());
+
+                        let it = (new_variables.indices()).zip(elems.iter_variables());
+                        for (target_index, var_index) in it {
+                            let var = env.source[var_index];
+                            let copy_var = storage_copy_var_to_help(env, var);
+                            env.target.variables[target_index] = copy_var;
+                        }
+
+                        let elem_index_start = env.target.tuple_elem_indices.len() as u32;
+
+                        // TODO: introduce a DenseTupleElems type, so we don't have to
```

Comment:
not needed now but you could encode this pretty easily on the current representation. make tuple_elem_indices_start a result; in the left case, it is the start of the tuple elems indices, in the right case it's the number of indices

Reply:
Great idea - I left a comment to that effect and renamed the struct to just `TupleElems`.